### PR TITLE
Fix SPI function to support INSERT, UPDATE and DELETE

### PR DESCRIPTION
**Pull request by HaozhouWang**
1. Fix backend to support non-returning INSERT, UPDATE and DELETE
2. Fix python client to support non-returning INSERT, UPDATE and DELETE
   and will return the number of rows that are processed.

### CODE REVIEW

File: src/common/comm_messages.c [hunk 1]
```diff
@@ -127,9 +127,14 @@ void free_result(plcMsgResult *res, bool isSender) {
         }
         free_type(&res->types[i]);
     }
-    pfree(res->types);
-    pfree(res->names);
 
+	if (res->types != NULL){
+		pfree(res->types);
```
Comment (zhangh43) [inline src/common/comm_messages.c]: better also set res->types=NULL

**Comment by paul-guo-** [comment]
+1

File: src/common/comm_messages.c (hunk 1 above)
Comment (HaozhouWang) [reply src/common/comm_messages.c]: Thanks, fixed

**Comment by zhangh43** [comment]
+1

**Pull request merged**